Documentation/Installation/EnvironmentConfiguration.rst

```diff
@@ -2,35 +2,197 @@
+https://github.com/vlucas/phpdotenv/ and parse it in your :file:`system/additional.php`
+
+You can require these libraries through Packagist/Composer or by including the dependency
+in your project manually (legacy installation).
+
+Example for `symfony/dotenv`:
+
+..  literalinclude:: Environments/_dotenv-symfony.php
+    :language: php
+    :caption: <config>/system/additional.php
+
+Example for `vlucas/phpdotenv`:
+
+..  literalinclude:: Environments/_dotenv-vlucas.php
+    :language: php
+    :caption: <config>/system/additional.php
+
+Once this code has loaded the :php:`$_ENV` variables from the specified file,
+you can access contents of the variables anywhere you need.
+
+.. _environment-helhum-dotenv:
+
+helhum/dotenv-connect
+---------------------
+
+You can also use https://github.com/helhum/dotenv-connector/ (via the Packagist
+package `helhum/dotenv-connector`) to allow accessing :php:`$_ENV` variables
+directly within the Composer autoload process.
+
+This has two nice benefits:
+
+*   You can even set the `TYPO3_CONTEXT` application context environment variable
+    through an :file:`.env` file, and no longer need to specify that in your webserver
+    configuration (e.g. via `.htaccess` or VirtualHost-Configuration).
+*   You do not need to add and maintain such loading code to your :file:`additional.php`
+    file.
+
+The drawback is that you'll have an additional dependency on another package, and
+glue code that is outside of your own implementation.
+
+.. _environment-phpconfig:
+
+Plain PHP configuration files
+=============================
+
+If the concept of requiring a specific file format and their loader dependencies
+seems like too much overhead for you, something similar can be achieved
+by including environment-specific :php:`.php` files
+
+For example, you can create a custom file like :file:`system/environment.php` that
+will only be placed on your specific target server (and not be kept in your versioning
+control system).
+
+..  literalinclude:: Environments/_environment.php
+    :language: php
+    :caption: <config>/system/environment.php
+
+This file would also need to be loaded through the additional configuration
+workflow:
+
+..  literalinclude:: Environments/_additional-native.php
+    :language: php
+    :caption: <config>/system/additional.php
+
+You can of course move such a file to a special :file:`Shared/Data/` directory
+(see :ref:`deploytypo3`), as long as you take care the file is outside
+your public web root directory scope.
+
+The file :file:`additional.php` can still contain custom changes that shall
+be applied to every environment of yours, and that is not managed through
+:file:`settings.php`.
+
+..  hint::
+    The file :file:`settings.php` is used by TYPO3 to store changes made through
+    the GUI of the backend. :file:`additional.php` always has the higher
+    priority, so configuration values there will overwrite the GUI configuration.
+
+..  tip::
```

Comment:
Like discussed, please remove this part as it may entangle users and this section should be more about a best practise.